src/SIL.XForge.Scripture/Services/SFProjectService.cs

```diff
@@ -84,11 +84,16 @@ public async Task<string> CreateProjectAsync(string curUserId, SFProjectCreateSe
         if (!userSecretAttempt.TryResult(out UserSecret userSecret))
             throw new DataNotFoundException("The user does not exist.");
 
+        string projectDir = Path.Combine(SiteOptions.Value.SiteDir, "sync", settings.ParatextId);
+        if (FileSystemService.DirectoryExists(projectDir))
+            throw new InvalidOperationException("A directory for this project already exists.");
```

Comment:
@pmachapman Is using an InvalidOperationException here okay or do you believe we should use a different exception? Currently the InvalidOperationException is caught in the SFProjectsRpcController.cs to present a clear message to the front end.

Reply:
Are you wanting the error to be displayed?

I don't think it is too out of place - contextually it makes sense compared to the other API endpoints to throw an `InvalidOperationException` for this type of problem.

You can always use a different exception code, convert it to a custom CommandErrorCode, then handle appropriately on the frontend, if you want to do something different.

Reply:
Yes. We ran into an issue when this was being tested and displaying the error message will make it easier to verify the issue and help resolve it.